scanner/mappers/mappers.go

```diff
@@ -22,6 +23,16 @@ var (
 		claircore.High:       v4.VulnerabilityReport_Vulnerability_SEVERITY_IMPORTANT,
 		claircore.Critical:   v4.VulnerabilityReport_Vulnerability_SEVERITY_CRITICAL,
 	}
+	// vulnNamePatterns is a prioritized list of regexes to match against
+	// vulnerability information to extract their ID.
+	vulnNamePatterns = []*regexp.Regexp{
+		regexp.MustCompile(`((RHSA|RHBA|RHEA)-\d{4}:\d+)|(ALAS[A-Z]*-\d{4}-\d+)`),
```

Comment:
`ALAS\d*` (ALAS, ALAS2, ALAS2023, etc)

Reply:
Can you point out to me some examples of ALA2*

Reply:
Moving the my previous question here, which is related:

> I think it makes sense to prioritize RHSA/RHBA/RHEA over CVE. CVE over GHSA. Why ALAS over CVE?
>
> Also, please provide the reference where you found ALAS2 and ALAS2023 (without a hyphen).

Reply:
Here's an example partial entry:

```
<update status="final" version="1.4" author="linux-security@amazon.com" type="security" from="linux-security@amazon.com"><id>ALAS2023-2023-001</id><title>Amazon Linux 2023 - ALAS2023-2023-001: Important priority package update for rsyslog</title><issued date="2023-02-17 20:41:00" /><updated date="2023-02-22 23:22:00" /><severity>Important</severity><description>Package updates are available for Amazon Linux 2023 that fix the following vulnerabilities:
CVE-2022-24903:
	A flaw was found in rsyslog's reception TCP modules. This flaw allows an attacker to craft a malicious message leading to a heap-based buffer overflow. This issue allows the attacker to corrupt or access data stored in memory, leading to a denial of service in the rsyslog or possible remote code execution.

CVE-2014-3634:
	A flaw was found in the way rsyslog handled invalid log message priority values. In certain configurations, a local attacker, or a remote attacker able to connect to the rsyslog port, could use this flaw to crash the rsyslog daemon or, potentially in rsyslog 7.x, execute arbitrary code as the user running the rsyslog daemon.
</description><references><reference href="http://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2014-3634" title="" id="CVE-2014-3634" type="cve" /><reference href="http://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2022-24903" title="" id="CVE-2022-24903" type="cve" /></references>
```

You can find more by following this path:

* Choose which Amazon version you care about between 2018.03, 2, 2023
* Depending on the version you have to follow a different link:
    * http://repo.us-west-2.amazonaws.com/2018.03/updates/x86_64/mirror.list
    * https://cdn.amazonlinux.com/2/core/latest/x86_64/mirror.list
    * https://cdn.amazonlinux.com/al2023/core/mirrors/latest/x86_64/mirror.list
* Go to that link and copy the URL inside the file. We'll call it `<URL>`
* Go to `<URL>/repodata/updateinfo.xml.gz`
* Open the file and you'll find the related vulns

Reply:
ClairCore uses the `<id>` field which in the given example would be `ALAS2023-2023-001`

Reply:
> I think it makes sense to prioritize RHSA/RHBA/RHEA over CVE. CVE over GHSA. Why ALAS over CVE?
That's just what we have now, and I believe ALAS is very similar to RHSA in that each ALAS may resolve one or more CVEs. The one I sent above [`ALAS2023-2023-001`](https://alas.aws.amazon.com/AL2023/ALAS-2023-001.html) seems to resolve two.

I'd say prefer it over CVE for now and we should address it around the time we address RHSA vs CVE